Set TNS_ADMIN env var to /etc, requires additional RPM

amaltaro opened the pull request:
This patch assumes that the node where this package will get installed, already has the `oracle-instantclient-tnsnames.ora` RPM installed. The latter is provided by CERN and automatically synchronizes the `tnsnames.ora` from AFS/EOS in the localhost (every hour). That said, we'd have to install that RPM via puppet on all the CMSWEB backends (are the frontends talking to oracle somehow?), on every vocms/production host and maybe on CRAB TaskWorker as well(?)

On your private CERN openstack VM, you can install this RPM as:
```
sudo yum install oracle-instantclient-tnsnames.ora --enablerepo={cernonly,cernonly-testing}
```

@smuzaffar @mrodozov I'm not sure this is the best way to address this issue.
@h4d4 you might have concerns as well.

h4d4 commented:
+1 to d27b31f77e08ffb09394cb029e8246c7062fc56a on slc7_amd64_gcc630.<br>Build log can be found <a href="https://cern.ch/cms-dmwm-builds/4885/test_d27b31f-20190416-163007.log" target="_blank">here</a>.

amaltaro commented:
@smuzaffar Shahzad, do these changes make sense to you (meaning that any service requiring this package is supposed to have the `tnsnames.ora` file under /etc, instead of the one shipped with the RPM)?

amaltaro commented:
I had to update the `cx-oracle` python specs because it was failing to build with these changes, error was:
```
* You can choose a different installation directory, i.e., one that is
on PYTHONPATH or supports .pth files

* You can add the installation directory to the PYTHONPATH environment
variable.  (It must then also be on PYTHONPATH whenever you run
Python and want to use the package(s) you are installing.)

* You can set up the installation directory to support ".pth" files by
using one of the approaches described here:

https://setuptools.readthedocs.io/en/latest/easy_install.html#custom-installation-locations


Please make the appropriate changes for your system and try again.
error: Bad exit status from /build/amaltaro/w/tmp/rpm-tmp.0djnpo (%install)
```

@smuzaffar can you have another look just to make sure I got it right?

h4d4 commented:
+1 to 97940e039ddd3afa4ed962f82af25779786811b5 on slc7_amd64_gcc630.<br>Build log can be found <a href="https://cern.ch/cms-dmwm-builds/4885/test_97940e0-20190503-110512.log" target="_blank">here</a>.

smuzaffar commented on oracle-env.spec:
```diff
@@ -1,7 +1,12 @@
+### Alan on 16/Apr/2019
+# Instead of using an out-dated tnsnames.ora - shipped with this package - it assumes
+# the node has this RPM installed "oracle-instantclient-tnsnames.ora", which will sync
+# tnsnames from EOS/AFS to the localhost, under /etc/ directory
```
@amaltaro , please remove `Source1: oracle-tnsnames.ora` and `cp %_sourcedir/oracle-tnsnames.ora* %i/etc/tnsnames.ora` lines too

smuzaffar commented:
@amaltaro , looks good. As I mentioned you need to clean up `oracle-tnsnames.ora` ref from oracle-env.spec

amaltaro commented:
Thanks for the review, Shahzad. I updated the spec according to your suggestions and also deleted the oracle-tnsnames.ora.file file. I'm now rebuilding the RPMs once again to make sure it works.

h4d4 commented:
+1 to 6b927fd44950217057411a8cca10ea7ae1e56391 on slc7_amd64_gcc630.<br>Build log can be found <a href="https://cern.ch/cms-dmwm-builds/4885/test_6b927fd-20190503-153511.log" target="_blank">here</a>.

amaltaro commented:
I also managed to build and upload comp RPMs.
Lina @h4d4 from my side it's ready to go, but as we discussed, I don't mind if it gets merged after the cmsweb production upgrade (I'll only need it for WMAgent anyways).

h4d4 commented:
@amaltaro Alan, 

thanks for taking care of this. The PR will be included into next release.

Best Regards, Lina.

yuyiguo commented:
@h4d4 @amaltaro 

How do I get the rpm? Is it included in the HG1905g? the oracle-env.spec already pointed to /etc. Will I get it deploy it automatically with DBS builded with HG1905g and above?

amaltaro commented:
@yuyiguo , I tried to install it on my private VM and I see no way to get it installed. Apparently it's only available for puppet managed nodes. The repo that it comes from
`From repo   : cernonly-testing`
is not even available and I can't resolve the url to it.

Given that we were using a 4-years old tnsnames.ora, I think we can simply download it once from one of the sources I mentioned in this JIRA ticket:
https://its.cern.ch/jira/browse/CMSVOC-59

and place it under `/etc`, and that's all that is needed.

yuyiguo commented:
@amaltaro 
Alan, I installed HG905g from comp on my VM and manually copied to replace the old one in /data/srv/current/sw/slc7_amd64_gcc630/cms/oracle-env/31/etc .

So I can just put a new one in /ect/tnsnames.ora, then I don't need to copy to /data/srv/current/sw/slc7_amd64_gcc630/cms/oracle-env/31/etc each time when I deploy a new release, right?

Thanks,
Yuyi

amaltaro commented:
> So I can just put a new one in /ect/tnsnames.ora, ..

that's correct, Yuyi. Once this PR gets merged and you start deploying RPMs using these new changes, `TNS_ADMIN` will be pointing to `/etc`, so you simply copy it under /etc and don't bother about it anymore.

Anyways, I created a ticket to see whether we can install that RPM in our openstack VMs, that would be the best option to keep an up-to-date copy in our dev nodes as well.

amaltaro commented:
@yuyiguo I got a reply from CERN IT and managed to install that RPM. The command can be seen above in my first comment/description of this PR. Please let me know if you have issues installing it.

yuyiguo commented:
@amaltaro 
Thanks Alan. I will try.

amaltaro commented:
@h4d4 Lina, now that the CMSWEB production upgrade is behind us, can we consider merging this PR? It will be needed in the beginning of the next week for a new WMAgent deployment campaign.

h4d4 commented:
@amaltaro Alan, 

yes there are no show-stoppers to merge it. I've just merged.

Best Regards, Lina.

amaltaro commented:
Perfect! Thanks!
